### PR TITLE
[binary_tree_partition] [1/2]: Add mincut helper functions and introduce `_binary_tree_partition_inds`

### PR DESCRIPTION
**Pull request by LinjianMa**
1. Overload `GraphsFlows.mincut` and `mincut_partitions`
2. Add `inds_binary_tree` and `inds_mps_order`. For a given network, the first one outputs a binary tree of uncontract inds, and the second one outputs an inds order that can be used to approximate the network by an MPS.

### CODE REVIEW

**Comment by LinjianMa** [comment]
@mtfishman I have made changes based on our last discussion. Could you review this?

File: src/utils.jl [hunk 1]
```diff
@@ -14,3 +14,24 @@ maybe_only(x::Tuple{T}) where {T} = only(x)
...
+  end
+  if length(line) <= 2
+    return line
+  end
+  return [line_to_tree(line[1:(end - 1)]), line[end]]
+end
+
+# vectorize a tree
+# example: [[1,2], [3,4]] = [1, 2, 3, 4]
+function vectorize(tree)
```
Comment (mtfishman) [inline src/utils.jl:31]: There is `collect(Leaves(tree))` from `AbstractTrees.jl` for this.

File: src/mincut.jl [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    newt = ITensor(inds1..., inds2...)
+    maxweight_tn[v] = newt
+  end
+  return maxweight_tn, out_to_maxweight_ind
+end
+
+"""
+Given a tn and outinds (a subset of noncommoninds of tn),
+get a binary tree structure of outinds based on given algorithm (mincut or mps).
+If algorithm="mps", the binary tree will have a line/mps structure.
```
Comment (mtfishman) [inline src/mincut.jl:83]: Looks like `algorithm` was replaced with `maximally_unbalanced`.

File: src/utils.jl (hunk 1 above)
Comment (LinjianMa) [reply src/utils.jl]: Done, removed this function

File: src/mincut.jl (hunk 2 above)
Comment (LinjianMa) [reply src/mincut.jl]: Done

**Comment by LinjianMa** [comment]
@mtfishman could you approve the workflow for this PR?

**Comment by mtfishman** [comment]
Looks good, thanks! I like the new function names.

**Pull request merged**